libraries/SRV_Channel/SRV_Channel_aux.cpp

```diff
@@ -248,6 +248,27 @@ void SRV_Channels::set_digital_outputs(uint16_t dig_mask, uint16_t rev_mask) {
     digital_mask |= dig_mask;
     reversible_mask |= rev_mask;
 
+    // add in NeoPixel and ProfiLED functions to digital array to determine anything else
+    // that should be disabled
+    for (uint8_t i = 0; i < NUM_SERVO_CHANNELS; i++) {
+        SRV_Channel &c = channels[i];
+        switch (c.function.get()) {
+            case SRV_Channel::k_LED_neopixel1:
+            case SRV_Channel::k_LED_neopixel2:
+            case SRV_Channel::k_LED_neopixel3:
+            case SRV_Channel::k_LED_neopixel4:
+            case SRV_Channel::k_ProfiLED_1:
+            case SRV_Channel::k_ProfiLED_2:
+            case SRV_Channel::k_ProfiLED_3:
+            case SRV_Channel::k_ProfiLED_Clock:
+                dig_mask |= 1U<<c.ch_num;
+                break;
+            default:
+                break;
+        }
+    }
+    disabled_mask = hal.rcout->get_disabled_channels(dig_mask);
```

Comment:
It is perhaps an existing issues but it seems digital_mask is only added to using the dig_mask argument... which makes me think that if a user changes the dig_mask somehow (presumably by modifying BLH_ parameter) both the old and new channels will be marked as digital until the next reboot.

Is this because AP_FETtecOneWire might also call this set_digital_outputs() method?... or maybe this method is called many times from BLHeli?  It doesn't seem to me that either is the case. I suspect that FETTec can't be used at the same time as BLHeli and BLHeli only calls this function once from within init().

Reply:
Right, existing issue - to fix probably needs a much more significant rewrite